File: Axiom/Internal/Ast/CallExpression.cs
using System.Collections.Generic;

namespace Axiom.Internal.Ast
{
    internal class CallExpression : Expression
    {
        public CallExpression(Expression callee, IList<Expression> arguments)
        {
            Callee = callee;
            Arguments = arguments;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public Expression Callee { get; private set; }
        public IList<Expression> Arguments { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/CallExpression.java
package com.moybl.axiom.ast;

import java.util.List;

public class CallExpression implements Expression {

	private Expression callee;
	private List<Expression> arguments;

	public CallExpression(Expression callee, List<Expression> arguments) {
		this.callee = callee;
		this.arguments = arguments;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Expression getCallee() {
		return callee;
	}

	public List<Expression> getArguments() {
		return arguments;
	}

}
